src/main/java/com/getindata/connectors/http/internal/table/lookup/HttpLookupTableSource.java

```diff
@@ -46,17 +51,20 @@ public class HttpLookupTableSource
     private final DynamicTableFactory.Context dynamicTableFactoryContext;
 
     private final DecodingFormat<DeserializationSchema<RowData>> decodingFormat;
+    @Nullable
+    private final LookupCache cache;
 
     public HttpLookupTableSource(
             DataType physicalRowDataType,
             HttpLookupConfig lookupConfig,
             DecodingFormat<DeserializationSchema<RowData>> decodingFormat,
-            DynamicTableFactory.Context dynamicTablecontext) {
-
+            DynamicTableFactory.Context dynamicTablecontext,
+            @Nullable LookupCache cache) {
```

Comment:
nit:
Im still not super happy that we can pass null here like described in @davidradl [PR](https://github.com/getindata/flink-http-connector/pull/94#discussion_r1636997742) but maybe its just me trying to over complicate things.

Reply:
The same approach is used in flink-connector-jdbc: https://github.com/apache/flink-connector-jdbc/blob/v3.2/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcDynamicTableSource.java#L85